packages/coverage-c8/src/provider.ts

```diff
@@ -29,6 +29,18 @@ export class C8CoverageProvider implements CoverageProvider {
   }
 
   onBeforeFilesRun() {
+    let tmpExist = existsSync(this.options.tempDirectory)
+
+    // clean tmp dir before run
+    if (tmpExist) {
+      rmSync(this.options.tempDirectory, { recursive: true, force: true })
+      tmpExist = false
+    }
```

Comment:
There should be no need to re-implement this logic. It is already done in `clean` function.

https://github.com/vitest-dev/vitest/blob/5b190e8490092f835240b60112fbef449c1af801/packages/coverage-c8/src/provider.ts#L35-L41

This function should be called from `packages/vitest` as it was before `0.22` release. I think it broke somewhere in https://github.com/vitest-dev/vitest/pull/1676.

Reply:
I'm not sure whether `onBeforeFilesRun` supports calling async `clean`. If support, I can change `onBeforeFilesRun` to call async `clean`. Besides, I think it's better to handle `clean` logic inside `C8CoverageProvider`.

Reply:
Or we can call `C8CoverageProvider.clean` in `initCoverageProvider`

https://github.com/vitest-dev/vitest/blob/5b190e8490092f835240b60112fbef449c1af801/packages/vitest/src/node/core.ts#L119-L127

Reply:
Coverage provider is instructed to clean here.

https://github.com/vitest-dev/vitest/blob/5b190e8490092f835240b60112fbef449c1af801/packages/vitest/src/node/core.ts#L108

But it seems that at this point `this.coverageProvider` has not yet been initialized at this point. This is clearly a bug. 

The `this.coverageProvider.clean()` should be moved so that it is called after `this.initCoverageProvider()`. Maybe here into `start`?

https://github.com/vitest-dev/vitest/blob/5b190e8490092f835240b60112fbef449c1af801/packages/vitest/src/node/core.ts#L148-L151

Reply:
Alright. I will modify my commit code

Reply:
done @AriPerkkio